File: solutions/python/resistor-color-duo/1/resistor_color_duo.py
BAND_COLOR = [
    "black",
    "brown",
    "red",
    "orange",
    "yellow",
    "green",
    "blue",
    "violet",
    "grey",
    "white"
]

def value(colors):
    color1 = colors[0]
    color2 = colors[1]

    first = BAND_COLOR.index(color1)
    second = BAND_COLOR.index(color2)

    return first * 10 + second

File: solutions/typescript/resistor-color-duo/1/resistor-color-duo.ts
const colorCodes: Record<string, number> = {
  black: 0,
  brown: 1,
  red: 2,
  orange: 3,
  yellow: 4,
  green: 5,
  blue: 6,
  violet: 7,
  grey: 8,
  white: 9,
};

export function decodedValue(colors: string[]): number {
  return Number(
    colors
      .slice(0, 2)
      .map(color => colorCodes[color])
      .join("")
  );
}